internal/services/nginx/nginx_configuration_resource.go

```diff
@@ -0,0 +1,318 @@
+	req := nginxconfiguration.NginxConfiguration{
+		Name: pointer.FromString(c.Name),
+		Properties: &nginxconfiguration.NginxConfigurationProperties{
+			RootFile: pointer.FromString(c.RootFile),
+		},
+	}
+
+	var files []nginxconfiguration.NginxConfigurationFile
+	for _, file := range c.ConfigFile {
+		files = append(files, nginxconfiguration.NginxConfigurationFile{
+			Content:     pointer.FromString(file.Content),
+			VirtualPath: pointer.FromString(file.VirtualPath),
+		})
+	}
+	req.Properties.Files = &files
+
+	if len(c.ProtectedFile) > 0 {
+		var protectedFiles []nginxconfiguration.NginxConfigurationFile
+		for _, file := range c.ProtectedFile {
+			protectedFiles = append(protectedFiles, nginxconfiguration.NginxConfigurationFile{
+				Content:     pointer.FromString(file.Content),
+				VirtualPath: pointer.FromString(file.VirtualPath),
+			})
+		}
+		req.Properties.ProtectedFiles = &protectedFiles
+	}
+
+	if c.PackageData != "" {
+		req.Properties.Package = &nginxconfiguration.NginxConfigurationPackage{
+			Data: pointer.FromString(c.PackageData),
+		}
+	}
+
+	return req
+}
+
+type ConfigurationResource struct{}
+
+var _ sdk.Resource = (*ConfigurationResource)(nil)
+
+func (m ConfigurationResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"nginx_deployment_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"config_file": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"content": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsBase64,
+					},
+
+					"virtual_path": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+				},
+			},
+		},
+
+		"protected_file": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"content": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsBase64,
+					},
+
+					"virtual_path": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+				},
+			},
+		},
+
+		"package_data": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"root_file": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+	}
+}
+
+func (m ConfigurationResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		// name of nginx configuration set to a fix value `default` by service team.
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
```

Comment:
there's probably not much point exposing this field? we have a few resources where `default` is the name and we intentionally don't expose it (since it's not something that's user configurable)